src/PrestaShopBundle/Twig/Component/QuickAccess.php

```diff
@@ -28,14 +28,198 @@
 
 namespace PrestaShopBundle\Twig\Component;
 
-use Link;
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use PrestaShop\PrestaShop\Core\Domain\Language\ValueObject\LanguageId;
+use PrestaShop\PrestaShop\Core\FeatureFlag\FeatureFlagSettings;
+use PrestaShop\PrestaShop\Core\FeatureFlag\FeatureFlagStateCheckerInterface;
+use PrestaShop\PrestaShop\Core\QuickAccess\QuickAccessRepositoryInterface;
+use PrestaShop\PrestaShop\Core\Util\Url\UrlCleaner;
+use PrestaShopBundle\Entity\Repository\TabRepository;
+use PrestaShopBundle\Service\DataProvider\UserProvider;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Security\Csrf\CsrfTokenManagerInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
 use Symfony\UX\TwigComponent\Attribute\AsTwigComponent;
 
 #[AsTwigComponent(template: '@PrestaShop/Admin/Component/Layout/quick_access.html.twig')]
 class QuickAccess
 {
-    public array $quickAccess;
-    public Link $link;
-    public string $quickAccessCurrentLinkIcon;
-    public string $quickAccessCurrentLinkName;
+    /**
+     * link to new product creation form
+     */
+    private const NEW_PRODUCT_LINK = 'index.php/sell/catalog/products/new';
+
+    /**
+     * link to new product creation form for product v2
+     */
+    private const NEW_PRODUCT_V2_LINK = 'index.php/sell/catalog/products-v2/create';
+
+    /**
+     * List of Quick Accesses to display
+     */
+    private array $quickAccesses = [];
+
+    /**
+     * Current Quick access by current request uri
+     */
+    private array|false|null $currentQuickAccess = null;
+
+    /**
+     * Clean current Url
+     */
+    private ?string $currentUrl = null;
+
+    /**
+     * Current url title
+     */
+    private ?string $currentUrlTitle = null;
+
+    /**
+     * Tokenized Urls cache
+     */
+    private array $tokenizedUrls = [];
+
+    public function __construct(
+        private readonly LegacyContext $context,
+        private readonly QuickAccessRepositoryInterface $quickAccessRepository,
+        private readonly TabRepository $tabRepository,
+        private readonly CsrfTokenManagerInterface $tokenManager,
+        private readonly UserProvider $userProvider,
+        private readonly RequestStack $requestStack,
+        private readonly FeatureFlagStateCheckerInterface $featureFlagStateChecker,
+        private readonly TranslatorInterface $translator,
+    ) {
+    }
+
+    /**
+     * Get quick accesses to display
+     */
+    public function getQuickAccesses(): array
+    {
+        if (empty($this->quickAccesses)) {
```

Comment:
I think you should test if null instead of empty because if the result of the following treatment is `[]`, the lazy loading won't work.